graphql_compiler/schema_generation/orientdb/get_orientdb_schema_graph.py

```diff
@@ -0,0 +1,488 @@
+                                                         schema introspection code, e.g.,
+                                                         '{}' for the embedded set type. Note
+                                                         that if the property is a collection
+                                                         type, it must have a default value.
+
+    Returns:
+        fully-constructed SchemaGraph object
+    """
+
+    toposorted_schema_data = toposort_classes(schema_data)
+
+    inheritance_sets = _get_inheritance_sets_from_schema_data(toposorted_schema_data)
+
+    class_name_to_definition = {
+        class_definition['name']: class_definition
+        for class_definition in toposorted_schema_data
+    }
+
+    non_graph_elements = _get_non_graph_elements(class_name_to_definition, inheritance_sets)
+    inner_collection_objs = _get_graphql_rep_of_non_graph_elements(
+        non_graph_elements, inheritance_sets)
+
+    elements = non_graph_elements
+    elements.update(
+        _get_edge_elements(class_name_to_definition, inheritance_sets, inner_collection_objs))
+    elements.update(
+        _get_vertex_elements(class_name_to_definition, inheritance_sets, inner_collection_objs))
+
+    # Initialize the connections that show which schema classes can be connected to
+    # which other schema classes, then freeze all schema elements.
+    _link_vertex_and_edge_types(elements, inheritance_sets)
+    for element in six.itervalues(elements):
+        element.freeze()
+
+    return SchemaGraph(elements, inheritance_sets)
+
+
+def _get_inheritance_sets_from_schema_data(schema_data):
+    """Return the inheritance sets from the OrientDB schema data."""
+    # For each class name, construct its inheritance set:
+    # itself + the set of class names from which it inherits.
+    inheritance_sets = dict()
+    for class_definition in schema_data:
+        class_name = class_definition['name']
+        immediate_superclass_names = get_superclasses_from_class_definition(
+            class_definition)
+
+        inheritance_set = set(immediate_superclass_names)
+        inheritance_set.add(class_name)
+
+        # Since the input data must be in topological order, the superclasses of
+        # the current class should have already been processed.
+        # A KeyError on the following line would mean that the input
+        # was not topologically sorted.
+        inheritance_set.update(chain.from_iterable(
+            inheritance_sets[superclass_name]
+            for superclass_name in immediate_superclass_names
+        ))
+
+        # Freeze the inheritance set so it can't ever be modified again.
+        inheritance_sets[class_name] = frozenset(inheritance_set)
+    return inheritance_sets
+
+
+def get_superclasses_from_class_definition(class_definition):
+    """Extract a list of all superclass names from a class definition dict."""
```

Comment:
Can we decide on one of `inheritance_set` and `superclass_set`? I prefer superclass, because inheritance goes both ways.

Reply:
This seems good to me. However, I'll do this in a following PR. I want to keep this diff as much as copy paste as possible.